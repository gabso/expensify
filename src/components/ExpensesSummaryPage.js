import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';


export const ExpensesSummaryPage = ({expensesAmount,expensesTotal}) => {

    const expensesWord = expensesAmount === 1 ? 'expense' : 'expenses';
    const expensesTotalFormated = numeral(expensesTotal / 100).format('$0,0.00');
    return (
<div>
    <h1>viewing {expensesAmount} {expensesWord} totaling {expensesTotalFormated}</h1>
    </div>
    );
};


const mapStateToProps = (state) => {

    const expenses = selectExpenses(state.expenses, state.filters);
    return {
        expensesAmount : expenses.length ,
      expensesTotal : selectExpensesTotal(expenses)
    };
  };
  
  export default connect(mapStateToProps)(ExpensesSummaryPage);
  