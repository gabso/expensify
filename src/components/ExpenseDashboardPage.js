import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import ExpensesSummaryPage from './ExpensesSummaryPage';

const ExpenseDashboardPage = () => (
  <div>
    <ExpensesSummaryPage/>
    <ExpenseListFilters />
    <ExpenseList />
  </div>
);

export default ExpenseDashboardPage;
