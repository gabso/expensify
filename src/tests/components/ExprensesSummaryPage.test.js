import React from 'react';
import { shallow } from 'enzyme';
import expenses from '../fixtures/expenses';
import {ExpensesSummaryPage} from '../../components/ExpensesSummaryPage';

test('should render ExpensesSummaryPage correctly with 1 report tatling 1.95$', () => {
  const wrapper = shallow(<ExpensesSummaryPage expensesAmount={1} expensesTotal={195} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummaryPage correctly with 2 report tatling 1140$', () => {
    const wrapper = shallow(<ExpensesSummaryPage expensesAmount={2} expensesTotal={114000} />);
    expect(wrapper).toMatchSnapshot();
  });
