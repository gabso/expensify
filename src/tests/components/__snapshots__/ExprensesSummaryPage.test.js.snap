// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummaryPage correctly with 1 report tatling 1.95$ 1`] = `
<div>
  <h1>
    viewing 
    1
     
    expense
     totaling 
    $1.95
  </h1>
</div>
`;

exports[`should render ExpensesSummaryPage correctly with 2 report tatling 1140$ 1`] = `
<div>
  <h1>
    viewing 
    2
     
    expenses
     totaling 
    $1,140.00
  </h1>
</div>
`;
